QEfficient/utils/cache.py

```diff
@@ -0,0 +1,41 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c) 2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# ----------------------------------------------------------------------------
+
+import json
+import os
+from pathlib import Path
+
+QEFF_HOME: Path = None
+if "QEFF_HOME" in os.environ:
+    QEFF_HOME = Path(os.environ["QEFF_HOME"])
+elif "XDG_CACHE_HOME" in os.environ:
+    QEFF_HOME = Path(os.environ["XDG_CACHE_HOME"]) / "qeff_models"
+else:
+    QEFF_HOME = Path("~/.cache/qeff_models").expanduser()
```

Comment:
Where is this used? check `utils/constants/QEFF_MODELS_DIR`

Reply:
I'll have to pick the new changes introduced in the other PR. I'll refactor to move it from constants.py to this one, as this will contains all utilities related to caching.

Reply:
I'll be doing this refactor when I move the export utils to base classes, as this will return in pathlib.Path, whereas the other one returns a string.

Reply:
This needs to be updated before we do code freeze. This is duplicate code.